packages/web-pkg/src/components/FilesList/ResourceTable.vue

```diff
@@ -996,59 +997,100 @@ export default defineComponent({
        */
       this.$emit('fileClick', { space, resources: [resource] })
     },
-    isResourceClickable({ id, status }: Resource) {
+    isResourceClickable(resource: Resource) {
       if (!this.areResourcesClickable) {
         return false
       }
 
       // TODO: remove as soon as pending & declined shares are accessible
-      if (status === ShareStatus.pending || status === ShareStatus.declined) {
+      if (
+        isShareResource(resource) &&
+        (resource.status === ShareStatus.pending || resource.status === ShareStatus.declined)
+      ) {
         return false
       }
 
-      return !this.disabledResources.includes(id)
+      return !this.disabledResources.includes(resource.id)
     },
-    getResourceCheckboxLabel(resource) {
+    getResourceCheckboxLabel(resource: Resource) {
       if (resource.type === 'folder') {
         return this.$gettext('Select folder')
       }
       return this.$gettext('Select file')
     },
-    getSharedWithAvatarDescription(resource) {
+    getSharedWithAvatarDescription(resource: Resource) {
+      if (!isShareResource(resource)) {
+        return
+      }
+
       const resourceType =
         resource.type === 'folder' ? this.$gettext('folder') : this.$gettext('file')
-      const shareCount = resource.sharedWith.filter((u) => !u.link).length
-      const linkCount = resource.sharedWith.filter((u) => !!u.link).length
+
```

Comment:
The following lines might be a bit complicated to read. 

The previous implementation basically accumulated people and link shares into one description. But there is no scenario where we need the description for both at the same time because we're either looking at a people share resource, or a link share resource ("Shared by me" <> "Sharing via link"). As a result we can simply this here.